earth_enterprise/src/fusion/autoingest/sysman/AssetD.h

```diff
@@ -56,4 +62,26 @@ class AssetImplD : public virtual AssetImpl
 typedef DerivedAssetHandle_<Asset, AssetImplD> AssetD;
 typedef MutableAssetHandleD_<AssetD> MutableAssetD;
 
+template<typename MutableAssetHandleType, typename ConfigType, typename MutableVersionHandleType>
```

Comment:
Asset versions already have an AssetType field that specifies their corresponding asset type. You can use that to eliminate the need to pass both the asset and asset version type in these two functions.

Reply:
Not sure that helps in this situation since one of the parameters is an asset type, so we can't have a parameter that's `MutableVersionHandleType::AssetType`.